server/commands/list_objects.go

```diff
@@ -0,0 +1,279 @@
+package commands
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/tuple"
+	serverErrors "github.com/openfga/openfga/server/errors"
+	"github.com/openfga/openfga/storage"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	MaximumConcurrentChecks = 100 // todo(jon-whit): make this configurable, but for now limit to 100 concurrent checks
+)
+
+type ListObjectsQuery struct {
+	Datastore             storage.OpenFGADatastore
+	Logger                logger.Logger
+	Tracer                trace.Tracer
+	Meter                 metric.Meter
+	ListObjectsDeadline   time.Duration
+	ListObjectsMaxResults uint32
+	ResolveNodeLimit      uint32
+}
+
+// Execute the ListObjectsQuery, returning a list of object IDs
+func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjectsRequest) (*openfgapb.ListObjectsResponse, error) {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return nil, err
+	}
+
+	allResults, _ := q.Meter.SyncInt64().Counter(
+		"openfga.listobjects.allresults",
```

Comment:
The `openfga.` prefix should be redundant. We should be setting a service name prefix when we instantiate our metrics collector anyways.

Can we use camelCase on `listObjects`?

And instead of `allresults` can we name this something more like `objectCount` or something like that?

Reply:
> The `openfga.` prefix should be redundant.

I kept it the same as the other metrics 🤷‍♀️ the prefix makes it faster to see all metrics available

Reply:
I'm not debating whether a prefix is helpful or not. Prefixes are definitely valuable. I'm saying that the prefix should be included in all metrics under the metric collected, because we should be setting ServiceName at the collector level and not per metric.

All I'm saying is that this should be refactored so that we don't have to prefix every metric with the service name. That should be a top-level thing.

Reply:
Ahhh ok. Separate PR :)